zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -448,11 +449,20 @@ where
     R: Copy + Debug,
     D: WalletWrite<Error = E, TxRef = R> + WalletWriteTransparent<UtxoRef = U>,
 {
-    // Check that the ExtendedSpendingKey we have been given corresponds to the
-    // ExtendedFullViewingKey for the account we are spending from.
-    if !wallet_db.is_valid_account_extfvk(account, extfvk)? {
-        return Err(E::from(Error::InvalidExtSk(account)));
-    }
+    // Obtain the UFVK for the specified account & use its internal change address
+    // as the destination for shielded funds.
+    let shielding_address = wallet_db
+        .get_unified_full_viewing_keys()
+        .and_then(|ufvks| {
+            ufvks
+                .get(&account)
+                .ok_or_else(|| E::from(Error::AccountNotFound(account)))
+                .and_then(|ufvk| {
+                    ufvk.sapling()
```

Comment:
Add a TODO here (with an issue that is blocked by #614) to have this formally select the most preferred shielded receiver. Currently that is Sapling because we can't receive or spend Orchard funds, but that will change soon and we want to remember to update this.

Reply:
Added #617